### PR TITLE
Add support for Finalized motion message

### PR DESCRIPTION
**Pull request by Julianahlk**
Test with [colonyCDapp#472](https://github.com/JoinColony/colonyCDapp/pull/472)

Adds support for the following motion messages:

- MotionFinalized

Contributes to https://github.com/JoinColony/colonyCDapp/issues/388

### CODE REVIEW

File: src/handlers/motions/motionVoteRevealed/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { BigNumber } from 'ethers';
+
+import { MotionMessage, MotionEvents } from '~types';
+
+interface Props {
```
Comment (jakubcolony) [inline src/handlers/motions/motionVoteRevealed/helpers.ts:5]: Props sound a bit odd in a non-React context. How about `Options`, `Params`?

File: src/handlers/motions/motionFinalized/motionFinalized.ts [hunk 2]
```diff
@@ -60,10 +63,19 @@ export default async (event: ContractEvent): Promise<void> => {
       ),
     );
 
+    const updatedMessages = [
+      ...messages,
+      {
+        name: MotionEvents.MotionFinalized,
+        messageKey: `${transactionHash}${logIndex}`,
```
Comment (jakubcolony) [inline src/handlers/motions/motionFinalized/motionFinalized.ts:70]: This formatting seems to be repeated in a few places. You could consider adding a util function to generate it

**Review by jakubcolony — APPROVED** [review]
Nice work 👍

File: src/handlers/motions/motionVoteRevealed/helpers.ts (hunk 1 above)
Comment (Julianahlk) [reply src/handlers/motions/motionVoteRevealed/helpers.ts:5]: Params sounds good

File: src/handlers/motions/motionFinalized/motionFinalized.ts (hunk 2 above)
Comment (Julianahlk) [reply src/handlers/motions/motionFinalized/motionFinalized.ts:70]: @jakubcolony you mean like this?
`export const getMessageKey = (
  transactionHash: string,
  logIndex: number,
): string => {
  return `${transactionHash}${logIndex}`;
};`

File: src/handlers/motions/motionVoteRevealed/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  messages,
+  messageKey,
+  voter,
+  votes: [nayVotes, yayVotes],
+}: Props): MotionMessage[] => {
+  const updatedMessages = [
+    ...messages,
+  ];
+
+  if (nayVotes.gte(yayVotes)) {
```
Comment (willm30) [inline src/handlers/motions/motionVoteRevealed/helpers.ts:22]: I would sync these with the motion state via the lamda, else it's going to publish this message every time someone reveals their vote.

**Review by willm30 — CHANGES_REQUESTED** [review]

File: src/handlers/motions/motionVoteRevealed/helpers.ts (hunk 3 above)
Comment (Julianahlk) [reply src/handlers/motions/motionVoteRevealed/helpers.ts]: Moved to cdapp fetchMotionState lambda

**Comment by Julianahlk** [comment]
@jakubcolony @willm30 
Based on will's comments I made significant changes to this PR and the associated #472 so they could use a re-review.

**Review by willm30 — APPROVED** [review]
Tested on front end and all is LGTM

**Review by jakubcolony — APPROVED** [review]
Good to go 👍

**Pull request merged**